sdk/core/core-http/src/serializer.ts

```diff
@@ -6,12 +6,448 @@ import * as base64 from "./util/base64";
+   */
+  required?: boolean;
+  /**
+   * Whether or not the current property allows mull as a value
+   */
+  nullable?: boolean;
+  /**
+   * The name to use when serializing
+   */
+  serializedName?: string;
+  /**
+   * Type of the mapper
+   */
+  // eslint-disable-next-line no-use-before-define
+  type: MapperType;
+  /**
+   * Default value when one is not explicitly provided
+   */
+  defaultValue?: any;
+  /**
+   * Constraints to test the current value against
+   */
+  constraints?: MapperConstraints;
+}
+
+export type MapperType =
+  | SimpleMapperType
+  | CompositeMapperType
+  | SequenceMapperType
+  | DictionaryMapperType
+  | EnumMapperType;
+
+export interface CompositeMapper extends BaseMapper {
+  type: CompositeMapperType;
+}
+
+export interface SequenceMapper extends BaseMapper {
+  type: SequenceMapperType;
+}
+
+export interface DictionaryMapper extends BaseMapper {
+  type: DictionaryMapperType;
+  headerCollectionPrefix?: string;
+}
+
+export interface EnumMapper extends BaseMapper {
+  type: EnumMapperType;
+}
+
+export type Mapper = BaseMapper | CompositeMapper | SequenceMapper | DictionaryMapper | EnumMapper;
+
+export interface UrlParameterValue {
+  value: string;
+  skipUrlEncoding: boolean;
+}
+
+// TODO: why is this here?
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export function serializeObject(toSerialize: any): any {
+  if (toSerialize == undefined) return undefined;
+  if (toSerialize instanceof Uint8Array) {
+    toSerialize = base64.encodeByteArray(toSerialize);
+    return toSerialize;
+  } else if (toSerialize instanceof Date) {
+    return toSerialize.toISOString();
+  } else if (Array.isArray(toSerialize)) {
+    const array = [];
+    for (let i = 0; i < toSerialize.length; i++) {
+      array.push(serializeObject(toSerialize[i]));
+    }
+    return array;
+  } else if (typeof toSerialize === "object") {
+    const dictionary: { [key: string]: any } = {};
+    for (const property in toSerialize) {
+      dictionary[property] = serializeObject(toSerialize[property]);
+    }
+    return dictionary;
+  }
+  return toSerialize;
+}
+
+/**
+ * Utility function to create a K:V from a list of strings
+ */
+function strEnum<T extends string>(o: Array<T>): { [K in T]: K } {
+  const result: any = {};
+  for (const key of o) {
+    result[key] = key;
+  }
+  return result;
+}
+
+// eslint-disable-next-line no-redeclare
+export const MapperType = strEnum([
+  "Base64Url",
+  "Boolean",
+  "ByteArray",
+  "Composite",
+  "Date",
+  "DateTime",
+  "DateTimeRfc1123",
+  "Dictionary",
+  "Enum",
+  "Number",
+  "Object",
+  "Sequence",
+  "String",
+  "Stream",
+  "TimeSpan",
+  "UnixTime"
+]);
+
 export class Serializer {
   constructor(
     public readonly modelMappers: { [key: string]: any } = {},
     public readonly isXML?: boolean
   ) {}
 
+  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
```

Comment:
maybe we can use `unknown` more in this file, but serializer is a hot mess, so I don't blame you for avoiding touching it. 😉